objax/optimizer/momentum.py

```diff
@@ -38,19 +38,21 @@ def __init__(self, vc: VarCollection, momentum: float = 0.9, nesterov: bool = Fa
         self.train_vars = ModuleList(TrainRef(x) for x in vc.subset(TrainVar))
         self.m = ModuleList(StateVar(jn.zeros_like(x.value)) for x in self.train_vars)
 
-    def __call__(self, lr: float, grads: List[jn.ndarray]):
+    def __call__(self, lr: float, grads: List[jn.ndarray], momentum: Optional[float] = None):
         """Updates variables and other state based on momentum (or Nesterov) SGD.
 
         Args:
            lr: the learning rate.
            grads: the gradients to apply.
+           momentum: optional, override the default momentum.
         """
         assert len(grads) == len(self.train_vars), 'Expecting as many gradients as trainable variables'
+        momentum = momentum or self.momentum
```

Comment:
Could it be the `or` confuses JAX (explained the error you saw with tracers)?
What if instead we did something like:
```
if momentum is None:
    momentum = self.momentum
```

Reply:
Yes, it worked.